posthog/session_recordings/session_recording_api.py

```diff
@@ -793,6 +796,85 @@ def bulk_delete(self, request: request.Request, *args: Any, **kwargs: Any) -> Re
             {"success": True, "deleted_count": deleted_count, "total_requested": len(session_recording_ids)}
         )
 
+    @extend_schema(exclude=True)
+    @action(methods=["POST"], detail=False, url_path="bulk_viewed")
+    def bulk_viewed(self, request: request.Request, *args: Any, **kwargs: Any) -> Response:
+        """Bulk mark recordings as viewed by providing a list of recording IDs."""
+
+        session_recording_ids = request.data.get("session_recording_ids", [])
+
+        if not session_recording_ids or not isinstance(session_recording_ids, list):
+            raise exceptions.ValidationError("session_recording_ids must be provided as a non-empty array")
+
+        if len(session_recording_ids) > MAX_RECORDINGS_PER_BULK_ACTION:
+            raise exceptions.ValidationError(
+                f"Cannot process more than {MAX_RECORDINGS_PER_BULK_ACTION} recordings at once"
+            )
+
+        user = cast(User, request.user)
+
+        # Create SessionRecordingViewed records for all session_recording_ids
+        # ignore_conflicts=True handles duplicates efficiently using the unique_together constraint
+        session_recordings_viewed_to_create = [
+            SessionRecordingViewed(
+                team=self.team,
+                user=user,
+                session_id=session_id,
+            )
+            for session_id in session_recording_ids
+        ]
+
+        created_records = SessionRecordingViewed.objects.bulk_create(
+            session_recordings_viewed_to_create, ignore_conflicts=True
+        )
+
+        viewed_count = len(created_records)
+
+        logger.info(
+            "bulk_recordings_viewed",
+            team_id=self.team.id,
+            user_id=user.id,
+            viewed_count=viewed_count,
+            total_requested=len(session_recording_ids),
+        )
+
+        return Response({"success": True, "viewed_count": viewed_count, "total_requested": len(session_recording_ids)})
+
+    @extend_schema(exclude=True)
+    @action(methods=["POST"], detail=False, url_path="bulk_not_viewed")
+    def bulk_not_viewed(self, request: request.Request, *args: Any, **kwargs: Any) -> Response:
+        """Bulk mark recordings as not viewed by providing a list of recording IDs."""
+
+        session_recording_ids = request.data.get("session_recording_ids", [])
```

Comment:
is this something users have asked for?
i'm wondering if maybe we just don't offer this - one less button and all that
i can't think when i'd want to do it 🙈